src/main/java/io/pravega/connectors/flink/util/FlinkPravegaUtils.java

```diff
@@ -55,20 +55,17 @@ private FlinkPravegaUtils() {
     }
 
     /**
-     * Utility method that derives default reader name from stream and scope name.
+     * Utility method that derives the reader name from taskName, index and parallelism.
      *
-     * @param scope The destination streams' scope name.
-     * @param streamNames Set of stream to read, used to generate the reader name.
+     * @param taskName the original task name
+     * @param index the index of the subtask
+     * @param total the total parallelism of the subtask
      * @return the generated default reader name.
      */
-    public static String getDefaultReaderName(final String scope, final Set<String> streamNames) {
-        final String delimiter = "-";
-        final String reader = streamNames.stream().collect(Collectors.joining(delimiter)) + delimiter + scope;
-        int hash = 0;
-        for (int i = 0; i < reader.length(); i++) {
-            hash = reader.charAt(i) + (31 * hash);
-        }
-        return Integer.toString(hash);
+    public static String getReaderName(final String taskName, final int index, final int total) {
+        String readerName = "Pravega-reader-for-" + taskName + "-" + index + "-" + total;
```

Comment:
It's a small thing I guess, but does the readerId REALLY need "Pravega-reader-for-" prefix?  It's obviously a Pravega Reader :-)  Perhaps a better ID would be `flink-task-XYZ`

Reply:
Thanks for catching it. Fixed.